### ISSUE TITLE
Installer: Check if Local Admin Permissions are Present before starting Installation

### ISSUE THREAD

**Issue by elliott94** [nvaccess/nvda#10666]
### Is your feature request related to a problem? Please describe.

Currently, if choosing to install NVDA (either from the downloadable Launcher or directly from an existing portable version), local administrator permissions are required to the current workstation to allow installation - this is expected, due to the access that NVDA requires. However, NSIS currently allows a user to begin installation - even if admin perms don't exist for their user account. This results in installation failing to complete.

### Describe the solution you'd like

Ideally, NSIS should check the presence of admin permissions on the machine before installation begins and warn the user if necessary. This may ultimately result in less confusion as to why installation failed to complete, and would allow the installer to be relaunched with the required set of permissions - rather than guesswork on the user's part as to whether their account has the right set of permissions in the first place.

**Comment by leonardder** [comment]
I recall that it is pretty difficult for an app to find out whether a user has admin rights on a system. People who are installing something globally just should be concious that administrative rights are required.

**Comment by Adriani90** [comment]
I think this is not possible at all in Windows. Installation is shown for every software when you are not admin, but when it comes to writing files on the disk, the admin permission promt appears. At least in my case it works like this.
cc: @gerald-hartig, @seanbudd  any further insights? Can you triage this please?

**Comment by XLTechie** [comment]
I suppose the question is: are we asking for elevation as early in the process as possible? As I recall, we probably aren't, on the grounds that you shouldn't try to become elevated before you need it. But most standard installers ask for elevation immediately--that is likely the user expectation.
